### PR TITLE
chore: improve build script

### LINKED ISSUE #75: issues with installation instructions
Tried following your installation instructions starting here: https://github.com/estkme-group/lpac/blob/ee4b208186a9a53f918db7391616905d5a83a265/README.md?plain=1#L38 .

So far, I've noticed a few issues:
1. I needed to run `sudo apt update` first.
2. When running `./scripts/build.sh make` I get `./scripts/build.sh: line 53: zip: command not found`. The fix was to run `sudo apt install zip`.
3. No `output` directory exists after compiling. Instead there is a file `lpac-linux-aarch64.zip` that I then need to unzip somewhere else.

### PR DESCRIPTION
**Pull request by septs**
resolve #75

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -62,27 +62,31 @@ mingw)
...
     ;;
 *)
     echo "Usage: $0 {make,debian,mingw,woa-mingw,woa-zig}"
     exit 1
     ;;
 esac
 
+rm -rf "$WORKSPACE/build"
+mkdir "$WORKSPACE/build"
+cp -r "$BUILD"/* "$WORKSPACE/build"
```
Comment (AndySchroder) [inline scripts/build.sh:90]: why don't you move instead of copy if you are just going to delete below?

**Comment by AndySchroder** [comment]
Does this address the need for `sudo apt update` and `sudo apt install zip`?

**Comment by AndySchroder** [comment]
The need for `sudo apt update` is a separate issue

**Comment by AndySchroder** [comment]
But if zip is installed, then you have to zip the output and then the user will have to re-unzip it to actually use the compiled file? Seems a little weird way to do it to me.

**Comment by septs** [comment]
https://github.com/estkme-group/lpac/blob/ee4b208186a9a53f918db7391616905d5a83a265/.github/workflows/build.yaml#L27-L33

**Comment by septs** [comment]
> The need for `sudo apt update` is a separate issue

what does it mean? hint install package before need update apt database?

**Comment by AndySchroder** [comment]
> https://github.com/estkme-group/lpac/blob/ee4b208186a9a53f918db7391616905d5a83a265/.github/workflows/build.yaml#L27-L33

Not sure what you are referencing here, but it seems to me like zipping or not zipping the output should be a build flag and not just doing it if a zip program is installed or not installed.

**Comment by AndySchroder** [comment]
> > The need for `sudo apt update` is a separate issue
> 
> what does it mean? hint install package before need update apt database?

This is a command you normally run before running `sudo apt install`. It refreshes the repository database so that when you try to `sudo apt install` packages, it can actually find them and find the latest version. Usually you have to run it once a day. In my case I haven't run it in several days so `sudo apt install` errored out. Since you never know when the last time someone ran `sudo apt update`, installation instructions should always say to do it every time. It only wastes a few seconds redoing it and no other harm done.

File: scripts/build.sh [hunk 2]
```diff
@@ -62,27 +62,36 @@ mingw)
...
     exit 1
     ;;
 esac
 
+rm -rf "$WORKSPACE/build"
+mkdir "$WORKSPACE/build"
+if [ -d "$BUILD/output" ]; then
+    cp -r "$BUILD/output" "$WORKSPACE/build"
+fi
+if [ -n "$(find "$BUILD" -name 'lpac-*.deb')" ]; then
```
Comment (AndySchroder) [inline scripts/build.sh:93]: nice to know that a .deb package can be created, but how do I even create that? I don't see it listed anywhere in the installation instructions.

**Comment by AndySchroder** [comment]
> > > The need for `sudo apt update` is a separate issue
> > 
> > 
> > what does it mean? hint install package before need update apt database?
> 
> This is a command you normally run before running `sudo apt install`. It refreshes the repository database so that when you try to `sudo apt install` packages, it can actually find them and find the latest version. Usually you have to run it once a day. In my case I haven't run it in several days so `sudo apt install` errored out. Since you never know when the last time someone ran `sudo apt update`, installation instructions should always say to do it every time. It only wastes a few seconds redoing it and no other harm done.

For example, you show `sudo apt update` here https://github.com/estkme-group/lpac/blob/ee4b208186a9a53f918db7391616905d5a83a265/scripts/setup-debian.sh#L3 . However, I'm not sure what that script is actually for.

**Pull request closed without merging**